### PR TITLE
Improve error message for non-existent asset

### LINKED ISSUE #296: hard to parse error if flutter.yaml points to a file that doesn't exist.
Here's the flutter.yaml:

```
assets:
  - assets/sprites.json
  - assets/sprites.png
  - assets/test.json
```

Here's the error:

```
[/Users/sethladd/Downloads/px_template 2] flutter start
The null object does not have a getter 'compress'.

NoSuchMethodError: method not found: 'compress'
Receiver: null
Arguments: []
severe: 
Exception:
dart:core                                                       Object.noSuchMethod
package:archive/src/zip_encoder.dart 36                         ZipEncoder.encode
package:sky_tools/src/commands/build.dart 210                   BuildCommand.build.<async>
dart:async                                                      _Completer.completeError
package:sky_tools/src/commands/build.dart 220                   BuildCommand.build.<async>
dart:async                                                      _asyncThenWrapperHelper
package:sky_tools/src/commands/build.dart                       BuildCommand.build
package:sky_tools/src/commands/build.dart 160                   BuildCommand.buildInTempDir.<async>
dart:async                                                      _asyncThenWrapperHelper
package:sky_tools/src/commands/build.dart                       BuildCommand.buildInTempDir
package:sky_tools/src/commands/start.dart 80                    StartCommand.runInProject.<async>
dart:async                                                      _asyncThenWrapperHelper
package:sky_tools/src/commands/start.dart                       StartCommand.runInProject
```

The error is: I don't have a assets/test.json in my project.

Maybe we can update the error message to indicate which file didn't exist?

Thanks for taking a look!

### PR DESCRIPTION
**Pull request by abarth**
Now we print the name of the asset and the base directory where we looked for
the error instead of throwing a null pointer exception.

Fixes #296

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/build.dart [hunk 1]
```diff
@@ -198,8 +198,14 @@ class BuildCommand extends FlutterCommand {
       archive.addFile(_createSnapshotFile(snapshotPath));
     }
 
-    for (_Asset asset in assets)
-      archive.addFile(_createFile(asset.key, asset.base));
+    for (_Asset asset in assets) {
+      ArchiveFile file = _createFile(asset.key, asset.base);
+      if (file == null) {
+        stderr.writeln('Cannot find asset "${asset.key}" in directory "${path.absolute(asset.base)}".');
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/build.dart]: use the logging infrastructure?

**Comment by Hixie** [comment]
LGTM

File: packages/flutter_tools/lib/src/commands/build.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter_tools/lib/src/commands/build.dart]: @devoncarew 's recommendation is not to use the logging system for messages intended for humans.  I don't fully understand when we're supposed to use logging and when we're supposed to just write to stdout/stderr.

Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/build.dart]: I think we were using logging too much early in the life of this tool. Generally, I think all errors should be written to stderr regardless of the logging level. Status messages to the user should probably go to stdout, again, regardless of the logging level. And only diagnostic stuff should go to logging (`about to run process 'foo arg1 arg2'`). This would be for users to use when things go off the rails and they're trying to determine where in the process things are breaking. And we could probably just dial the use of logging back down to use `info`, so there's less guesswork about which things go where.

Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/build.dart]: Filed https://github.com/flutter/flutter/issues/461 to track this.
